File: Python/2433.py
# 2433. Find The Original Array of Prefix Xor
# https://leetcode.com/problems/find-the-original-array-of-prefix-xor/submissions/1327260397/

class Solution:
  def findArray(self, pref: List[int]) -> List[int]:
    out = [pref[0]]

    for i in range(1, len(pref)):
      out.append(pref[i] ^ pref[i - 1])

    return out

File: Rust/2433.rs
/**
 * 2433. Find The Original Array of Prefix Xor
 * https://leetcode.com/problems/find-the-original-array-of-prefix-xor/submissions/1327260397/
 */

impl Solution {
  pub fn find_array(pref: Vec<i32>) -> Vec<i32> {
    let mut out: Vec<i32> = Vec::new();
    out.push(pref[0]);

    for i in 1..pref.len() {
      out.push(pref[i] ^ pref[i - 1]);
    }

    return out;
  }
}
